src/Vault.ts

```diff
@@ -0,0 +1,419 @@
+                claimableRedeemCurrency: new Currency(maxWithdraw, investmentCurrency.decimals),
+                claimableRedeemSharesEquivalent: new Token(maxRedeem, shareCurrency.decimals),
+                pendingInvestCurrency: new Currency(pendingInvest, investmentCurrency.decimals),
+                pendingRedeemShares: new Token(pendingRedeem, shareCurrency.decimals),
+                claimableCancelInvestCurrency: new Currency(claimableCancelInvestCurrency, investmentCurrency.decimals),
+                claimableCancelRedeemShares: new Token(claimableCancelRedeemShares, shareCurrency.decimals),
+                hasPendingCancelInvestRequest,
+                hasPendingCancelRedeemRequest,
+                investmentCurrency,
+                shareCurrency,
+              }
+            }).pipe(
+              repeatOnEvents(
+                this._root,
+                {
+                  address: [this.address, restrictionManagerAddress],
+                  abi: [ABI.LiquidityPool, ABI.RestrictionManager],
+                  eventName: [
+                    'UpdateMember',
+                    'CancelDepositClaim',
+                    'CancelDepositClaimable',
+                    'CancelDepositRequest',
+                    'CancelRedeemClaim',
+                    'CancelRedeemClaimable',
+                    'CancelRedeemRequest',
+                    'Deposit',
+                    'DepositClaimable',
+                    'DepositRequest',
+                    'RedeemClaimable',
+                    'RedeemRequest',
+                    'Withdraw',
+                  ],
+                  filter: (events) =>
+                    events.some(
+                      (event) =>
+                        event.args.receiver?.toLowerCase() === address ||
+                        event.args.controller?.toLowerCase() === address ||
+                        event.args.sender?.toLowerCase() === address ||
+                        event.args.owner?.toLowerCase() === address ||
+                        // UpdateMember event
+                        (event.args.user?.toLowerCase() === address &&
+                          event.args.token?.toLowerCase() === shareCurrency.address)
+                    ),
+                },
+                this.chainId
+              )
+            ),
+          ])
+        ),
+        map(([currencyBalance, shareBalance, allowance, investment]) => ({
+          ...investment,
+          shareBalance: new Token(shareBalance.toBigInt(), investment.shareCurrency.decimals),
+          investmentCurrencyBalance: currencyBalance,
+          investmentCurrencyAllowance: allowance,
+        }))
+      )
+    )
+  }
+
+  /**
+   * Place an order to invest funds in the vault. If an order exists, it will increase the amount.
+   * @param investAmount - The amount to invest in the vault
+   */
+  increaseInvestOrder(investAmount: bigint | number) {
```

Comment:
Nit: param name here is `investAmount`, while for `increaseRedeemOrder` it is `shares`. 

Would either use `assets` here, or `redeemAmount` in `increaseRedeemOrder`